Content.Shared/_Impstation/Kodepiia/Components/KodepiiaConsumeActionComponent.cs

```diff
@@ -0,0 +1,28 @@
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared._Impstation.Kodepiia.Components;
+
+[RegisterComponent, NetworkedComponent]
+public sealed partial class KodepiiaConsumeActionComponent : Component
+{
+    [DataField]
+    public EntityUid? ConsumeAction;
+    [DataField]
+    public string? ConsumeActionId = "ActionKodepiiaConsume";
+
+    public List<SoundSpecifier?> SoundPool = new()
+    {
+        new SoundPathSpecifier("/Audio/Effects/gib1.ogg"),
+        new SoundPathSpecifier("/Audio/Effects/gib2.ogg"),
+        new SoundPathSpecifier("/Audio/Effects/gib3.ogg"),
+    };
+
+    [DataField(required: true)] [ViewVariables(VVAccess.ReadWrite)]
+    public DamageSpecifier Damage = new();
+
+    [DataField]
+    public bool Gib = true;
```

Comment:
really minor but maybe this should be `CanGib` or `ShouldGib`, i got scared reading this